File: github-miner/src/types/Issue.ts
import { Comment } from "./Comment";
import { Label } from "./Label";
import { User } from "./User";

export type Issue = {
  number?: number;
  title?: string;
  body?: string;
  state?: string;
  createdAt?: string;
  updatedAt?: string;
  closedAt?: string;
  createdBy?: User;
  comments?: Comment[];
  assignees?: User[];
  labels?: Label[];
};

File: GithubAnalyzer/Models/Issue.cs
namespace GithubAnalyzer.Models
{
    public class Issue
    {
        public int Number { get; set; }

        public int Title { get; set; }

        public string Body { get; set; }

        public string State { get; set; }

        public DateTime? CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        public DateTime? ClosedAt { get; set; }

        public Author CreatedBy { get; set; }

        public List<Comment> Comments { get; set; }

        public List<Author> Assignees { get; set; }

        public List<Label> Labels { get; set; }
    }
}
